Can't build on Manjaro/Arch

Running ./zcutil/build.sh -j$(nproc) fails on Manjaro 64bit

```
checking for boostlib >= 1.20.0... yes
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system-mt... no
configure: error: Could not link against boost_system-mt !
```

The zCash solution is to run build.sh w/ `--disable-proton`, but even though this is present in the configuration file, the script doesn't accept it, "invalid argument --disable-proton"
 **edit: I see that disable-proton is default on the build file.**

I'm currently attempting to change `PREFIX="/usr/"` to see if that works
update: somehow didn't pick up on the db headers.

Tried now with setting `--with-boost-libdir=/usr/lib64` still getting same error.

--- a/zcutil/build.sh
+++ b/zcutil/build.sh
@@ -100,12 +100,12 @@ fi
 # might break the entry above to find gcc- on arm
 if [ -f "/etc/arch-release" ]; then
     if [ -f "/usr/bin/gcc-5" ]; then
-    CC=gcc-5
-    CXX=g++-5
+      CC=gcc-5
+      CXX=g++-5
+    else
+      echo 'gcc5 required, please install using "sudo pacman -S gss5"'
+      exit 1
     fi
-else
-echo 'gcc5 required, please install using "sudo pacman -S gss5"'
-exit 1
 fi
 
 PREFIX="$(pwd)/depends/$BUILD/"

--- a/zcutil/build.sh
+++ b/zcutil/build.sh
@@ -96,6 +96,18 @@ then
     shift
 fi
 
+# Arch workaround for gcc 7
+# might break the entry above to find gcc- on arm
+if [ -f "/etc/arch-release" ]; then
+    if [ -f "/usr/bin/gcc-5" ]; then
+    CC=gcc-5
+    CXX=g++-5
+    fi
+else
+echo 'gcc5 required, please install using "sudo pacman -S gss5"'
+exit 1
+fi
+
 PREFIX="$(pwd)/depends/$BUILD/"
 
 eval "$MAKE" --version

